includes/blocks.inc

```diff
@@ -74,23 +74,88 @@ function islandora_basic_collection_clear_object_count_cache() {
  * Outputs a JSON string to the buffer.
  */
 function islandora_basic_collection_object_count_callback() {
+  $count_method = variable_get('islandora_basic_collection_count_method', 'SPARQL');
+
+
   $cid = islandora_basic_collection_get_object_count_block_cache_id();
 
   if ($value = cache_get($cid, 'cache')) {
     $substitutions = $value->data;
   }
   else {
+
     $tuque = islandora_get_tuque_connection();
     $objects_query_array = islandora_basic_collection_get_query_info(array(
       'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
       'collection_listing' => TRUE,
       'all_objects' => TRUE,
     ));
     $collection_query_array = islandora_basic_collection_get_query_info(array(
-      'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
-      'collection_listing' => TRUE,
-      'all_objects' => FALSE,
-    ));
+    'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
+    'collection_listing' => TRUE,
+    'all_objects' => FALSE,
+  ));
+
+
+  // Setup Solr query for collection and object counts.
+  if ($count_method == "Solr") {
+    $all_cmodels = islandora_get_content_models();
+    $models_to_exclude = variable_get('islandora_basic_collection_object_count_listing_content_models_to_restrict', FALSE);
+    $cmodels_to_count = array();
+    foreach($all_cmodels AS $cmodel) {
+      if ($models_to_exclude[$cmodel['pid']] != '0') {
+        $cmodels_to_count[] = $cmodel['pid'];
+      }
+    }
+
+    // Collection count Solr query.
+    $collection_query = "RELS_EXT_hasModel_uri_mt:collection";
+    $qp = new IslandoraSolrQueryProcessor();
```

Comment:
This creates a dependency on `islandora_solr_search`, which isn't accounted for... should we instead be defining a hook that defines the mechanism to generate the counts? Our RI mechanism can be broken out into an implementation of it in this module, and the Solr implementation implemented in `islandora_solr_search`?

Reply:
@adam-vessey i agree with the missing explicit dependency but I can also see a point made on making Solr a dependency? Do you guys have Islandora's without Solr running? It is not standard in islandora but there could be also a "bland" dependency by adding a module exist statement and documenting that IF sole search is enabled this would be available. Not perfect but as a way of encouraging ANY 7.x commits to actual happen.

Reply:
Feels like it would be a lot easier to define a dependency on `islandora_solr_search` here instead. Is there a reason not to do that?

Reply:
I like @DiegoPino's suggestion here -- could only present the Solr option if the Solr module exists, this way we avoid a dependency and also keep the code in the Collection module, which makes more sense than putting it in Solr.

Reply:
Sure, I see no issue with a soft dependency.

Reply:
Addressed with latest push - does it suit?

Reply:
👀 give me a minute